DogTales/src/components/physics.hpp

```diff
@@ -0,0 +1,28 @@
+#pragma once
+#include <bave/app.hpp>
+#include <bave/graphics/sprite.hpp>
+
+namespace component {
+
+static constexpr glm::vec2 max_velocity{1000.f, 1000.f};
+static constexpr float tick_limit{0.03f}; //this should possibly go somewhere else and be globally accessible
+
+class Physics {
+
+  public:
+
+	glm::vec2 m_position{};
+	glm::vec2 m_velocity{};
+	glm::vec2 m_acceleration{};
+	
+	glm::vec2 m_friction{};
+	float m_gravity{};
+	float m_mass{};
```

Comment:
If all this is public, a constructor isn't needed, users can just do:
```cpp
auto physics = Physics{
  .m_friction = /* ... */,
  .m_gravity = /* ... */,
  .m_mass = /* ... */,
};
```
As a style thing, this is one reason why I don't prefix `m_` for public members.

Reply:
However, shouldn't `gravity` be shared amongst all physics components, as opposed to being an individual member on each instance?

Reply:
do you think friction, position, velocity etc. should be public? or should I write getters for them?

Reply:
> I agree intuitively about gravity, but being able to change it might make it easier to simulate light-weightedness (for certain enemies or drops, etc).

Ah, nice idea! One way to streamline both: have a `static constexpr float gravity_v{10.0f}` in-class constant (or whatever default value), and initialize `m_gravity` to that: `float m_gravity{gravity_v}`. This way the gravity for each instance is customizable, and it takes the default value otherwise, liberating the user from having to initialize it on all instances.

Reply:
I think I'll go with all public for now, just to make it easier to write the collision stuff for whoever will be doing that

Reply:
this isn't working because it says a designator cannot be used with a non-aggregate type (even though I deleted the constructor). is my syntax wrong here?
```cpp
component::Physics m_physics = {.friction{0.999f, 0.999f}, .gravity{-1000.f}, .mass{1.f}};
```

Reply:
> this isn't working because it says a designator cannot be used with a non-aggregate type (even though I deleted the constructor). is my syntax wrong here?
> 
> ```c++
> component::Physics m_physics = {.friction{0.999f, 0.999f}, .gravity{-1000.f}, .mass{1.f}};
> ```

Is this a member variable? Try removing the `=` sign.

Reply:
yep, it's a member variable of `Player`. neither of those solutions worked either. here's the new `physics.hpp` if the issue lies there:

```cpp
namespace component {

constexpr glm::vec2 max_velocity_v{1000.f};
constexpr bave::Seconds tick_limit_v{100ms};

class Physics {

	bave::Seconds m_residue{};

  public:

	glm::vec2 position{};
	glm::vec2 velocity{};
	glm::vec2 acceleration{};

	glm::vec2 friction{};
	float gravity{};
	float mass{};

	void tick(bave::Seconds const dt);
	void integrate(bave::Seconds dt);
};

}
```

Reply:
ok I realized it's the `m_residue` being private

Reply:
so, I guess I need to define a constructor then, because `m_residue` being public feels pretty wrong